pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -353,14 +353,17 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 	out := &route.Route{
 		Match:    translateRouteMatch(match, node),
 		Metadata: util.BuildConfigInfoMetadata(virtualService.Meta),
+		Name:     virtualService.Name,
```

Comment:
what happens if there are multiple routes? does envoy complain?

Reply:
Good catch @rshriram , I did not think about that scenario.  I also worry about there being hidden character/length requirements in envoy for route names that we don't enforce for the names of our VirtualService objects.  We can not do random alpha-numeric generation, since that wouldn't produce consistent route names between XDS iterations and would break whatever name matching an EnvoyFilter is currently using.  Thoughts?

Reply:
Why don't you just set the name in the VS if you want to patch it?

Reply:
True, but we have the `route_name` as part of envoy's default access logs.  As soon as users start creating VirtualServices, that access log field goes to `-` which isn't very user-friendly either